### PR TITLE
ICU-20927 Adding French narrow duration unit test.

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20927
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [x] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

**Comment by srl295** [comment]
[CLDR-11543] but this is the ICU repo?

**Comment by sffc** [comment]
This was a CLDR issue and these are the ICU tests for that feature.

Should a new ICU ticket be used?

**Comment by srl295** [comment]
> Should a new ICU ticket be used?

for commits into ICU, yes.

**Comment by srl295** [comment]
@sffc ping?

File: icu4c/source/test/intltest/measfmttest.cpp [hunk 1]
```diff
@@ -3549,6 +3551,29 @@ void MeasureFormatTest::Test21060_AddressSanitizerProblem() {
...
+    Measure H5M10[] = {
+        {5, MeasureUnit::createHour(status), status},
+        {10, MeasureUnit::createMinute(status), status}
+    };
+    UnicodeString result;
+    FieldPosition pos;
+    mf.formatMeasures(H5M10, UPRV_LENGTHOF(H5M10), result, pos, status);
+    assertEquals("Should have consistent spacing", u"5h 10min", result);
+
+    // Test additional locales:
```
Comment (echeran) [inline icu4c/source/test/intltest/measfmttest.cpp:3566]: Is this a placeholder for future tests for other locales that use narrow-nbsp in their duration formatting?

Comment (sffc) [reply icu4c/source/test/intltest/measfmttest.cpp:3566]: It's intended as a way to spot-check for other languages that might have a similar problem as French.  Uncomment the lines and look at the output.  I could delete it, but I see no reason to.

**Comment by sffc** [comment]
> Was a Java port of this intended to be here, or is it coming later?

I don't think it's necessary to do a Java port, because this is testing a data issue, not a code issue.

**Pull request merged**